icepyx/core/APIformatting.py

```diff
@@ -181,12 +184,56 @@ def to_string(params):
     return "&".join(param_list)
 
 
+ParameterType = Literal["CMR", "required", "subset"]
+# DevGoal: When Python 3.12 is minimum supported version, migrate to PEP695 style
+T = TypeVar("T", bound=ParameterType)
```

Comment:
This is the grossest thing to me about the Python type system. The generics are much better in 3.12. Basically, we're going to use this to mark the `Parameter` function as varying on the `partype` parameter.